mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -185,6 +186,55 @@ void AbstractControllerExecution::setVelocityCmd(const geometry_msgs::TwistStamp
   // TODO so there should be no loss of information in the feedback stream
 }
 
+bool AbstractControllerExecution::checkCmdVelIgnored(const geometry_msgs::Twist& cmd_vel)
+{
+  // check if the velocity ignored check is enabled or not
+  if (cmd_vel_ignored_tolerance_ <= 0.0)
+  { 
+    return false;
+  }
+
+  const bool robot_stopped = robot_info_.isRobotStopped(1e-3, 1e-3);
+
+  // compute linear and angular velocity magnitude
+  const double cmd_linear = std::hypot(cmd_vel.linear.x, cmd_vel.linear.y);
+  const double cmd_angular = std::abs(cmd_vel.angular.z);
+
+  const bool cmd_is_zero = cmd_linear < 1e-3 && cmd_angular < 1e-3;
+
+  // velocity is not being ignored
+  if (!robot_stopped || cmd_is_zero)
+  {
+    if (!first_ignored_time_.is_zero())
+    {
+      first_ignored_time_ = ros::Time();
+    }
+    return false;
+  }
+
+  if (first_ignored_time_.is_zero())
+  {
+    // set first_ignored_time_ to now if it was zero
+    first_ignored_time_ = ros::Time::now();
+  }
+
+  const double ignored_duration = (ros::Time::now() - first_ignored_time_).toSec();
+  ROS_WARN_THROTTLE(1,
```

Comment:
S: maybe this should go after the error (line 234) so you don't show both WARN and ERROR together